src/battle_ai_util.c

```diff
@@ -434,6 +434,46 @@ bool32 IsDamageMoveUnusable(u32 move, u32 battlerAtk, u32 battlerDef)
         if (IsDynamaxed(battlerDef))
             return TRUE;
         break;
+    case EFFECT_HEAT_CRASH:
+        if (IsDynamaxed(battlerDef))
+            return TRUE;
+        break;
+    case EFFECT_OHKO:
+        if (IsDynamaxed(battlerDef))
+            return TRUE;
+        break;
+    case EFFECT_DESTINY_BOND:
+        if (IsDynamaxed(battlerDef))
+            return TRUE;
+        break;
+    case EFFECT_SKILL_SWAP:
+        if (IsDynamaxed(battlerDef))
+            return TRUE;
+        break;
+    case EFFECT_ENTRAINMENT:
+        if (IsDynamaxed(battlerDef))
+            return TRUE;
+        break;
+    case EFFECT_ROAR:
+        if (IsDynamaxed(battlerDef))
+            return TRUE;
+        break;
+    case EFFECT_DISABLE:
+        if (IsDynamaxed(battlerDef))
+            return TRUE;
+        break;
+    case EFFECT_ENCORE:
+        if (IsDynamaxed(battlerDef))
+            return TRUE;
+        break;
+    case EFFECT_TORMENT:
+        if (IsDynamaxed(battlerDef))
+            return TRUE;
+        break;
+    case EFFECT_INSTRUCT:
+        if (IsDynamaxed(battlerDef))
+            return TRUE;
+        break;
```

Comment:
This function only handles damaging moves so they will have to be added in `AI_CalcMoveScore`. Also you can just fall through.

Example:
```
    case EFFECT_LOW_KICK:
    case EFFECT_HEAT_CRASH:
        if (IsDynamaxed(battlerDef))
            return TRUE;
        break;
```